tesseract_core/sdk/api_parse.py

```diff
@@ -215,12 +217,26 @@ def validate_tesseract_api(src_dir: Path) -> None:
             )
 
         if obj.num_args is not None:
-            if _get_func_argnums(toplevel_objects[obj.name]) != obj.num_args:
+            func_argnames = _get_func_argnames(toplevel_objects[obj.name])
+            func_argnums = len(func_argnames)
+            if func_argnums != obj.num_args:
                 raise ValidationError(
-                    f"{obj.name} must have {obj.num_args} arguments: {obj.arg_names}.\n"
-                    f"However, {tesseract_api_location} specifies {len(toplevel_objects[obj.name].args.args)}  "
-                    f"arguments: {tuple(a.arg for a in toplevel_objects[obj.name].args.args)}."
+                    f"{obj.name} must have {obj.num_args} arguments: {', '.join(obj.arg_names)}.\n"
+                    f"However, {tesseract_api_location} specifies {func_argnums} "
+                    f"arguments: {', '.join(func_argnames)}."
                 )
+            else:
```

Comment:
Why not make this unconditional?

```suggestion

        if obj.arg_names is not None:
```

Reply:
I've removed the `else`, but don't see the need for `if obj.arg_names is not None`. We have already tested that `obj.arg_num is not None` and have no exceptions to the rule. If there is a need for an exception in the future (i.e. we decide to specify 0 arguments instead of `None`) I suggest we add this in then to avoid confusion now.